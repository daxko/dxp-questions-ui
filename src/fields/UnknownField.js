var React = require('react');

var UnknownField = React.createClass({

	render: function() {
		return (
			<div>UNKNOWN field type of {this.props.question.type}</div>
		);
	}

});

module.exports = UnknownField;
