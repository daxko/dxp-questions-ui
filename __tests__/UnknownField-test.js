import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import UnknownField from '../src/fields/UnknownField'
const json = require('../snippets/json/unknown_question.json')

configure({adapter: new Adapter() })

describe('UnknownField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<UnknownField question={json}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
