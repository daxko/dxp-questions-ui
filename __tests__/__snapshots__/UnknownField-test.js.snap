// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UnknownField
    question={
      Object {
        "description": "",
        "max_length": 50,
        "read_only": false,
        "required": false,
        "title": "This question will not be rendered.",
        "type": "nonexistent",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "UNKNOWN field type of ",
        "nonexistent",
      ],
    },
    "ref": null,
    "rendered": Array [
      "UNKNOWN field type of ",
      "nonexistent",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "UNKNOWN field type of ",
          "nonexistent",
        ],
      },
      "ref": null,
      "rendered": Array [
        "UNKNOWN field type of ",
        "nonexistent",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
